File: src/main/java/employees/java8/Restaurant.java
package employees.java8;

import java.util.HashSet;
import java.util.Optional;
import java.util.Set;

class Restaurant extends Building {

    Set<Employee> employees;

    Menu menu = new Menu();

    public Restaurant(int roomsCount, boolean hasKitchen, int bathroomsCount, Set<Employee> employees) {
        super(roomsCount, hasKitchen, bathroomsCount);
        this.employees = employees;
    }

    public Optional<Employee> findEmployeeByName(String fullName) {
        return employees.stream()
                .filter(employee -> employee.getFullName().equalsIgnoreCase(fullName))
                .findFirst();
    }

    public void add(Employee newEmployee) {
        employees.add(newEmployee);
    }

    public void add(MenuItem menuItem) {
        menu.add(menuItem);
    }

    public Optional<MenuItem> findMenuItemByName(String menuItemName) {
        return menu.findByName(menuItemName);
    }

    public Menu getMenu() {
        return menu;
    }

    public Set<Employee> getEmployees() {
        return employees;
    }
}

File: src/main/kotlin/employees/Restaurant.kt
package employees

import java.math.BigDecimal

class Restaurant(
        roomsCount: Int,
        hasKitchen: Boolean,
        bathroomsCount: Int,
        val employees: MutableSet<Employee> = mutableSetOf()
) : Building(roomsCount, hasKitchen, bathroomsCount) {


    val menu = Menu()

    fun findEmployeeByName(name: String) = employees.firstOrNull { name == it.fullName }

    fun add(employee: Employee) {
        employees.add(employee)
    }

    fun add(menuItem: MenuItem) {
        menu.add(menuItem)
    }

    fun findMenuItemByName(menuItemName: String): MenuItem? = menu.findByName(menuItemName)

    fun printMenu() {
        menu.print()
    }


}

class Menu {

    private val items = mutableSetOf<MenuItem>()

    fun add(menuItem: MenuItem) {
        items.add(menuItem)
    }

    fun findByName(name: String): MenuItem? = items.firstOrNull { name == it.name }

    fun print() {
        items.forEachIndexed { index, item -> println("$index. $item") }
    }

}

data class MenuItem(
        val name: String,
        val desc: String,
        val price: BigDecimal) {

    override fun toString() = "$name, $desc with price: ${price.toPlainString()}"

}